SYCL/Reduction/reduction_placeholder.cpp

```diff
@@ -58,28 +61,34 @@ void test(T Identity, size_t WGSize, size_t NWItems) {
   }
 }
 
+template <typename KernelName, typename T, int Dim, class BinaryOperation>
+void test(T Identity, T Init, size_t WGSize, size_t NWItems) {
+  testOneCase<KNGroup<KernelName, class RWCase>, T, Dim, BinaryOperation,
+              access::mode::read_write>(Identity, Init, WGSize, NWItems);
+  testOneCase<KNGroup<KernelName, class DWCase>, T, Dim, BinaryOperation,
+              access::mode::discard_write>(Identity, Init, WGSize, NWItems);
+}
+
 int main() {
   // fast atomics and fast reduce
-  test<class KernelName_LpplYqDbNN, int, 1, ONEAPI::plus<int>>(0, 49, 49 * 5);
-  test<class KernelName_FlDEESAfXYXiBZhnEDQ, int, 0, ONEAPI::plus<int>>(0, 8,
-                                                                        8);
+  test<class AtomicReduce1, int, 1, ONEAPI::plus<int>>(0, 77, 49, 49 * 5);
+  test<class AtomicReduce2, int, 0, ONEAPI::plus<int>>(0, -77, 8, 8);
 
   // fast atomics
-  test<class KernelName_caKErpdwXzEsGGkr, int, 0, ONEAPI::bit_or<int>>(0, 7,
-                                                                       7 * 3);
-  test<class KernelName_YbnKY, int, 1, ONEAPI::bit_or<int>>(0, 4, 128);
+  test<class Atomic1, int, 0, ONEAPI::bit_or<int>>(0, 233, 7, 7 * 3);
+  test<class Atomic2, int, 1, ONEAPI::bit_or<int>>(0, 177, 4, 128);
 
   // fast reduce
-  test<class KernelName_JQuAndqdovQbAHmVLm, float, 1, ONEAPI::minimum<float>>(
-      getMaximumFPValue<float>(), 5, 5 * 7);
-  test<class KernelName_MBbbTWwSc, float, 0, ONEAPI::maximum<float>>(
-      getMinimumFPValue<float>(), 4, 128);
+  test<class Reduce1, float, 1, ONEAPI::minimum<float>>(
+      getMaximumFPValue<float>(), -5.0, 5, 5 * 7);
+  test<class Reduce2, float, 0, ONEAPI::maximum<float>>(
+      getMinimumFPValue<float>(), -5.0, 4, 128);
 
   // generic algorithm
-  test<class KernelName_WpNdTbTtYt, int, 0, std::multiplies<int>>(1, 7, 7 * 5);
-  test<class KernelName_yAwH, int, 1, std::multiplies<int>>(1, 8, 16);
-  test<class KernelName_BNuHxeewzfXATi, CustomVec<short>, 0,
-       CustomVecPlus<short>>(CustomVec<short>(0), 8, 8 * 3);
+  test<class Generic1, int, 0, std::multiplies<int>>(1, 2, 7, 7 * 5);
+  test<class Generic2, int, 1, std::multiplies<int>>(1, 3, 8, 16);
+  test<class Generic13, CustomVec<short>, 0, CustomVecPlus<short>>(
```

Comment:
Naming scheme is better than it was but it is unclear why we have Generic13. Is it misprint?

Reply:
It is just a name. Generic here means that it does not use optimized versions with either fast-reduce or fast-atomic operations and uses generic tree-reduction algorithm + multiple kernels invocation inside.

Reply:
Fixed in e6df87c69e290cd2ad11b5466a98ed8ed8507d16